backend/types.bal

```diff
@@ -42,15 +43,99 @@ public type App record {|
     # Tag color of the target app
     string tagColor;
     # Icon asset name or key
-    string iconName;
+    string icon;
     # User who added the link
     string addedBy;
     # Whether the current user has favorited this link (0 = no, 1 = yes)
     int isFavourite;
 |};
 
-# Structure for Action Enum
+# [Database] Create App record.
+public type CreateApp record {|
+    # Display title
+    string header;
+
+    # Target URL
+    @constraint:String{
+        pattern: {
+            value: NON_EMPTY_URL,
+            message: "The URL should be non empty and valid URL"
+        }
+    }
+    string url;
+    
+    # Short description
+    string description;
+    # Version label of the target app
+    string versionName;
+    # Tag id of the target app
+    int tagId;
+    # Tag name of the target app
+    string tagName;
+
+    # Tag color of the target app
+    @constraint:String {
+        pattern: {
+            value: NON_EMPTY_HEX_VALUE,
+            message: "Color value should be a valid  hex value"
+        }
+    }
+    string tagColor;
+
+    # Icon asset name or key
+    @constraint:String{
+        pattern: {
+            value: NON_EMPTY_BASE64_STRING,
+            message: "icon must be base64 (optionally prefixed with data:image/svg+xml;base64"
+        }
+    }
+    string icon;
+
+    # User who added the link
+    string addedBy;
+    # User groups of the target app
+    string[] userGroups;
+    # Is the App is active or not
+    boolean isActive;
+|};
+
+# [Database] Tag record.
+public type Tag record {|
+    # Unique identifier of the tag
+    int id;
+    # Display name of the tag
+    string name;
+|};
+
+# User action for marking/unmarking apps as favourites.
 public enum Action {
+    # Mark an app as favourite
     FAVOURITE = "favourite",
+    # Remove an app from favourites
     UNFAVOURITE = "unfavourite"
```

Comment:
Shall we not use the term `unfavourite`? 
The meaning of unfavourite is not being a favorite; especially : being regarded with special **disfavor** or **dislike**. I think what we are trying to do here is just `remove favourite`. In other words we are not providing the ability to the user to mark apps that he/she dislikes. 
WDYT?
@Rashmika998 @yuk7hi @sacheeramesh @shayanmalinda